Assets/Scripts/Construction/PlayerConstructionHandler.cs

```diff
@@ -128,6 +136,19 @@ private Vector3 SnapToGrid(Vector3 vec, bool centerX, bool centerZ)
             return null;
         }
 
+        private void IsPlacementPositionValid()
+        {
+            Ray objectRay = new Ray(constructionPreview.transform.position, Vector3.down);
```

Comment:
It's still dependent on Construction Preview. Use currentPlacementPosition instead
```suggestion
            if (current.PlacementPosition == null) return false;
            Ray objectRay = new Ray(currentPlacementPosition.Value, Vector3.down);
```